Systematize clustbench run (software grid)

imallona opened the pull request:
Ongoing...

Incorporates part of https://github.com/btraven00/clustering_example/commit/e0989fb052f684850d5ac5dc96b2cd2b0423402c

imallona commented:
@imallona extend the makefile and the yaml-izer starting with conda on some versions, e.g. to reproduce Mark's singularity recipe results (with conda)

btraven00 commented:
I'm not sure what the merge plan is, but I'm sure there is one.

Just noting (possibly already known) that this PR is not able to run the whole clustbench example (missing some of the envs etc).

imallona commented:
@imallona add https://github.com/imallona/clustering_report/blob/6b4765d081f8882c7f7f40bf85fddefc02c1c7f2/quick-dirty-analysis.R

imallona commented:
FCPS instability is fixed (it was the seed) https://github.com/imallona/clustering_report/pull/3#issuecomment-2933828393

btraven00 commented on yamls/Clustering_apptainer.yml:
```diff
@@ -0,0 +1,211 @@
+# this file has been generated automatically - DO NOT EDIT BY HAND
```
I think I would remove this line; just because it's confusing if we're not including the generation mechanism in the repo. Perhaps we can link to the generation script once we have it merged into the paper repo.

btraven00 commented on yamls/Clustering_apptainer.yml:
```diff
@@ -0,0 +1,211 @@
+    description: "R with some plotting dependencies"
+    conda: envs/rmarkdown.yml
+    envmodule: rmarkdown/0.1.0-gfbf-2024a-r-4.4.2
+    apptainer: oras://quay.io/omnibenchmark/fcps:0.1.0
```
I'm not sure if you're intending to leave here pieces of the other PR for reference (that's my impression of this folder), but I'd stick to just one reference oras repository.

imallona replied:
thanks removed https://github.com/omnibenchmark/clustering_example/pull/30/commits/a88325db9473fb5f5c8eed4fc52490b8a92549ee

btraven00 commented on yamls/Clustering_apptainer_vanilla.yml:
```diff
@@ -1,42 +1,27 @@
-storage_api: S3
-storage_bucket_name: clusteringexampleapptainer
-software_backend: apptainer
+# this file has been generated automatically - DO NOT EDIT BY HAND
```
same. remove? (the files in yamls)

imallona replied:
thanks removed https://github.com/omnibenchmark/clustering_example/pull/30/commits/a88325db9473fb5f5c8eed4fc52490b8a92549ee

btraven00 reviewed (approved):
Good enough for me to merge. After merging this we can sync the paper repo and move the templating there, clarify the registries etc.

Left a couple comments, some stuff can be deleted for clarity I think.

Only one thing I didn't see: is there any ref to what's in the ORAS registry? shall we move to the quay one later on?

imallona commented:
The hardcl and softcl nondeterminism is somehow documented and some consistency checks have been added from https://github.com/imallona/clustbench_fcps/pull/4/commits/e8888d4b025b6e4848b1e749f956e9cf385edad3 on

btraven00 reviewed (changes requested):
* [ ] envs/README could use some update (remove reference to unexsisting env files)
* [ ] maybe add a script to do the easybuild builds?

imallona commented:
does it work ignoring checksums?

DanInci reviewed (approved):
Looks good to merge, apart from the conda env question left.

imallona commented:
Perhaps premature but it's running well for me, user `easybuild01`, tmux `imallona` if you'd like to keep an eye on it, or to retry with (full paths to ease browsing):

```bash
mkdir -p /home/easybuild01/imallona/src

## retrieve `develop` easyconfigs as for today, because Perl 5x CPAN errors 
##   were common in the past but issues have been fixed already (I hope)
##   please see https://github.com/easybuilders/easybuild-easyconfigs/issues?q=is%3Aissue%20Perl-bundle-CPAN
git clone https://github.com/easybuilders/easybuild-easyconfigs.git

## clone the clustering example repo - current state (`branch imallona`)
git clone git@github.com:omnibenchmark/clustering_example.git
cd clustering_example
git checkout imallona

## specify recent easyconfigs as robotspath hoping someone fixed the Perl 5 CPAN easyconfig (I think it's fixed)
eb --robot-paths=/home/easybuild01/imallona/src/easybuild-easyconfigs/easybuild/easyconfigs/ \
    --robot \
    /home/easybuild01/imallona/src/clustering_example/envs/clustbench.eb
```

imallona commented on envs/fcps.yml:
```diff
@@ -15,3 +15,12 @@ dependencies:
+  - bioconda::bioconductor-complexheatmap
+  - conda-forge::r-jsonlite
+  - conda-forge::r-dplyr
+  - conda-forge::r-r.utils
```
thanks https://github.com/omnibenchmark/clustering_example/pull/30/commits/65262c930dd1f431ce08d92646bb3cea7d6d6dfd

imallona commented:
Something's going on with pydantic-core (oras recipe), conda works but singularity doesn't anymore

btraven00 commented:
```== FAILED: Installation ended unsuccessfully: shell command 'make ...' failed with exit code 2 in extensions step for Perl-bundle-CPAN-5.38.0-GCCcore-13.2.0.eb (took 24 mins 19 secs)
== Results of the build can be found in the log file(s) /tmp/eb-nz9hamr2/easybuild-Perl-bundle-CPAN-5.38.0-20250623.040132.jTmpV.log
== Summary:
   * [SUCCESS] Doxygen/1.9.8-GCCcore-13.2.0
   * [SUCCESS] giflib/5.2.1-GCCcore-13.2.0
   * [SUCCESS] gzip/1.13-GCCcore-13.2.0
   * [SUCCESS] LittleCMS/2.15-GCCcore-13.2.0
   * [SUCCESS] lz4/1.9.4-GCCcore-13.2.0
   * [SUCCESS] groff/1.23.0-GCCcore-13.2.0
   * [SUCCESS] zstd/1.5.5-GCCcore-13.2.0
   * [FAILED]  Perl-bundle-CPAN/5.38.0-GCCcore-13.2.0
   * [SKIPPED] LibTIFF/4.6.0-GCCcore-13.2.0
   * [SKIPPED] libwebp/1.3.2-GCCcore-13.2.0
   * [SKIPPED] intltool/0.51.0-GCCcore-13.2.0
   * [SKIPPED] OpenJPEG/2.5.0-GCCcore-13.2.0
   * [SKIPPED] Pillow/10.2.0-GCCcore-13.2.0
   * [SKIPPED] gperf/3.1-GCCcore-13.2.0
   * [SKIPPED] util-linux/2.39-GCCcore-13.2.0
   * [SKIPPED] fontconfig/2.14.2-GCCcore-13.2.0
   * [SKIPPED] X11/20231019-GCCcore-13.2.0
   * [SKIPPED] Tk/8.6.13-GCCcore-13.2.0
   * [SKIPPED] Tkinter/3.11.5-GCCcore-13.2.0
   * [SKIPPED] matplotlib/3.8.2-gfbf-2023b
   * [SKIPPED] clustbench/1.6-foss-2023b
ERROR: Installation of Perl-bundle-CPAN-5.38.0-GCCcore-13.2.0.eb failed: "shell command 'make ...' failed with exit code 2 in extensions step for Perl-bundle-CPAN-5.38.0-GCCcore-13.2.0.eb"

(omnibenchmark) easybuild02@omnibenchmark:~/easybuild-easyconfigs$ git show
commit 78a0a64a53d921fcf3c3bc83ac66966626bfa15f (HEAD -> develop, origin/develop, origin/HEAD)
Merge: b2b6c0db2e 474fef3b3d
Author: Simon Branford <4967+branfosj@users.noreply.github.com>
Date:   Sat Jun 21 11:58:05 2025 +0100

    Merge pull request #23178 from SebastianAchilles/20250621104359_new_pr_Ninja1130

    {tools}[GCCcore/14.3.0] Ninja v1.13.0, Meson v1.8.2, libpciaccess v0.18.1

(omnibenchmark) easybuild02@omnibenchmark:~/easybuild-easyconfigs$```

imallona commented:
> Something's going on with pydantic-core (oras recipe), conda works but singularity doesn't anymore. Oddly enough it used to work with older omnibenchmark versions, same apptainer.

fixed https://github.com/omnibenchmark/clustering_example/pull/30/commits/1209b48afa2cbda3a6cb2fce4bd0d88640e9cea5 ; ci/cd also updated with https://github.com/omnibenchmark/clustering_example/pull/30/commits/0e3338ea2513cd72346ad623ec4d423c0932cedc

only runs with pydantic-core==2.34.1

imallona commented:
thanks @btraven00 , shall we meet to discuss and trigger some easybuilds together to figure out why do we get different outcomes?
